fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -872,6 +876,55 @@ protected OptExpression queryBasedRewrite(RewriteContext rewriteContext, ScalarO
                         new ReplaceColumnRefRewriter(queryExpression.getOp().getProjection().getColumnRefMap());
                 queryCompensationPredicate = rewriter.rewrite(queryCompensationPredicate);
             }
+            MaterializedView mv = mvRewriteContext.getMaterializationContext().getMv();
```

Comment:
put these in a function, and plz add some comment to desc what these does.

Reply:
done